Make necessary modifications to support guacscanner

jsf9k opened the pull request:
## 🗣 Description ##

This pull request makes the necessary modifications for the addition of the [cisagov/guacscanner](https://github.com/cisagov/guacscanner) functionality:
- Grant the Guacamole instance read-only access to EC2.
- Grant the Guacamole instance access the EC2 VPC endpoint.
- Remove the [cloud-init](https://cloud-init.io/) code for pre-creating Guacamole connections.
- Bump the IMDS hop limit to two for the Guacamole instance so that the Docker containers running on it can access IMDS.

## 💭 Motivation and context ##

The functionality of [cisagov/guacscanner](https://github.com/cisagov/guacscanner) is required for the RPT UAT taking place in the COOL.

See also:
- cisagov/guacscanner/pull/1
- cisagov/guacscanner-docker#2
- cisagov/guacamole-composition#30
- cisagov/ansible-role-guacamole#38
- cisagov/guacamole-packer#53

## 🧪 Testing ##

I have applied these changes to env0 in our staging COOL environment and verified that the modifications listed above all function as expected.

## ✅ Checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - _eschew scope creep!_
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
- [x] All new and existing tests pass.

jsf9k committed:
Add the EC2 read-only policy to the guacamole instance

This instance will be running the cisagov/guacscanner-docker Docker
image, which requires these permissions to identify EC2 instance for
which connections should be added to the Guacamole database.

jsf9k committed:
Remove code to create Guacamole connections

This is now handled in the Guacamole composition by the
cisagov/guacscanner-docker Docker container.

jsf9k committed:
Create a security group specifically for the EC2 interface endpoint

Use it to allow access to the Guacamole instance.  The Guacamole
instance has to be able to query EC2 for instance information in order
to keep the Guacamole connection database in sync as instances appear
and disappear.

jsf9k committed:
Bump up the IMDS hop limit

This is required so that the Docker containers running on the host can use IMDS.  It also requires adding a permission to the ProvisionAccount policy.

jsf9k committed:
Update README via terraform-docs

dav3r commented on guacamole_ec2.tf:
```diff
@@ -52,7 +53,7 @@ resource "aws_instance" "guacamole" {
     # default value).  This effectively disallows the retrieval of an
     # IMDSv2 token via this machine from anywhere else.
-    http_put_response_hop_limit = 1
+    http_put_response_hop_limit = 2
```
The comment above should be updated to explain why the hop limit has been bumped up to 2.

jsf9k replied:
Please see commit 0969c94.

dav3r replied:
Great comment! 📜

felddy replied:
Not for nothing... but does it make sense to run the container with `host` networking instead of `bridge` mode.  This would allow us to leave the IDMS hop limit at `1`.  

https://docs.docker.com/network/host/

I'm not asking to change this now, but it might be a discussion worthy of our collective time.

jsf9k replied:
@felddy, how would host networking work with respect to one container being able to see the other containers in the composition?  The `guacscanner` container needs to be able to communicate with the `postgres` container.

dav3r commented on guacamole_iam.tf:
```diff
@@ -37,6 +37,16 @@ resource "aws_iam_role_policy" "guacamole_assume_delegated_role_policy" {
 }
 
+# Attach the EC2 read-only policy to this role.  This policy is
+# required by the cisagov/guacamole-connection-scanner Docker image
```
Looks like this is the current name now:
```suggestion
# required by the cisagov/guacscanner Docker image
# (https://github.com/cisagov/guacscanner-docker)
```

dav3r reviewed (approved):
Nice - looks great!  🥑 🚀 
Please note the two minor thangs I commented on.

jsf9k committed:
Correct name of guacscanner Docker image

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

jsf9k committed:
Correct comment

The previous comment did not explain why we set the hop limit to two
in this case.

jsf9k committed:
Remove two input variables that are no longer used

felddy reviewed (approved):
🦾 🦃 

Looks great.  I did have a comment about the IDMS hop limit wrt Docker that might be worth talking about at some point.

mcdonnnj reviewed (approved):
💪💪💪
